src/main/java/touro/chess/PawnPiece.java

```diff
@@ -0,0 +1,75 @@
+package touro.chess;
+
+import touro.chess.AbstractPiece;
+import touro.chess.Location;
+import touro.chess.Move;
+import touro.chess.PieceColor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PawnPiece extends AbstractPiece {
+
+    public PawnPiece(Location location, PieceColor color) {
+        super(location, color);
+    }
+
+    @Override
+    public List<Move> getMoves() {
+        Location location = super.getLocation();
+        int row = location.getRow();
+        int col = location.getColumn();
+
+        List<Move> moves = new ArrayList<Move>();
+
+        if (getColor() == PieceColor.Black) {
```

Comment:
You can shorten this code by defining another variable based on the color.

```
int movement;
int startingRow;
if (getColor() == PieceColor.Black) {
movement = +1;
startingRow = 1;
}
else {
movement = -1;
startingRow = 6;
}
```

And then you should be able to get rid of what's in the if statements because they only differ on those variables.

Reply:
then what would I do with this?

Reply:
Try doing what I told you to do...

Reply:
I am trying, but sometimes `movement = - 2` or `movement = + 2`
Also, what about when I want to test if it's at 7 or 0, each color has its own requirements.

Reply:
You shouldn't need the `if (getColor() == PieceColor.Black)` conditional. Removing it should make things easier to read.

Reply:
I don't really understand where you are going with this... how can I do that?